### PR TITLE
Merge pai config

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: tools/nni_cmd/launcher_utils.py [hunk 1]
```diff
@@ -269,6 +273,9 @@ def validate_pai_trial_conifg(experiment_config):
             print_warning(warning_information.format('dataDir'))
         if experiment_config.get('trial').get('outputDir'):
             print_warning(warning_information.format('outputDir'))
+    if experiment_config.get('trainingServicePlatform') == 'pai':
+        if experiment_config.get('trial').get('paiConfigPath'):
+            get_yml_content(experiment_config['trial']['paiConfigPath'])
```
Comment (QuanluZhang) [inline tools/nni_cmd/launcher_utils.py]: what is the meaning of this line? unused return content

File: src/nni_manager/training_service/pai/paiK8S/paiK8STrainingService.ts [hunk 2]
```diff
@@ -185,7 +185,16 @@ class PAIK8STrainingService extends PAITrainingService {
             }
         }
 
-        return yaml.safeDump(paiJobConfig);
+        if (this.paiTrialConfig.paiConfigPath) {
+            try {
+                const additionalPAIConfig = yaml.safeLoad(fs.readFileSync(this.paiTrialConfig.paiConfigPath, 'utf8'));
+                return yaml.safeDump({...paiJobConfig, ...additionalPAIConfig});
+            } catch (error) {
+                this.log.error(`Loading ${this.paiTrialConfig.paiConfigPath} error: ${error}`);
```
Comment (QuanluZhang) [inline src/nni_manager/training_service/pai/paiK8S/paiK8STrainingService.ts]: this error message is not accurate, could be:
error occurs during loading and merge .... : ${error}

File: tools/nni_cmd/launcher_utils.py (hunk 1 above)
Comment (SparkSnail) [reply tools/nni_cmd/launcher_utils.py]: validate the format of yaml file

File: src/nni_manager/training_service/pai/paiK8S/paiK8STrainingService.ts (hunk 2 above)
Comment (SparkSnail) [reply src/nni_manager/training_service/pai/paiK8S/paiK8STrainingService.ts]: fixed.

File: src/nni_manager/package.json [hunk 3]
```diff
@@ -26,7 +26,8 @@
     "ts-deferred": "^1.0.4",
     "typescript-ioc": "^1.2.4",
     "typescript-string-operations": "^1.3.1",
-    "webhdfs": "^1.2.0"
+    "webhdfs": "^1.2.0",
+    "deepmerge": "^4.2.2"
```
Comment (liuzhe-lz) [inline src/nni_manager/package.json]: Looks like your are not using `yarn add`

Comment (SparkSnail) [reply src/nni_manager/package.json]: > Looks like your are not using `yarn add`

fixed.

File: src/nni_manager/package.json [hunk 4]
```diff
@@ -13,6 +13,7 @@
     "azure-storage": "^2.10.2",
     "chai-as-promised": "^7.1.1",
     "child-process-promise": "^2.2.1",
+    "deepmerge": "^4.2.2",
```
Comment (chicm-ms) [inline src/nni_manager/package.json]: there should be some updates in yarn.lock needs to be checked in.

Comment (SparkSnail) [reply src/nni_manager/package.json]: fixed

**Pull request merged**